File: hellojava/src/main/java/com/fathzer/oop/ACoolThing.java
package com.fathzer.oop;

public class ACoolThing extends AbstractThing implements CoolInterface {
    public ACoolThing(String name) {
        super(name);
    }

    @Override
    public void doSomething() {
        System.out.println(getName() + " does something");
    }

    @Override
    public void doSomethingCool() {
        System.out.println(getName() + " do something cool");
    }
}

File: hellogo/pkg/oop/a_cool_thing.go
package hellogo

import "fmt"

type ACoolThing struct {
    *AbstractThing
}

func NewACoolThing(name string) *ACoolThing {
    return &ACoolThing{
        AbstractThing: NewAbstractThing(name),
    }
}

func (t *ACoolThing) GetName() string {
    return t.AbstractThing.GetName()
}

func (t *ACoolThing) DoSomething() {
    fmt.Println(t.GetName() + " does something")
}

func (t *ACoolThing) DoSomethingCool() {
    fmt.Println(t.GetName() + " does something cool")
}
